packages/devextreme/js/ui/menu/ui.menu.js

```diff
@@ -269,6 +269,38 @@ class Menu extends MenuBase {
     _render() {
         super._render();
         this._initAdaptivity();
+        this._attachFocusOutHandler();
+    }
+
+    _isTargetOutOfComponent(relatedTarget) {
+        const $menuTarget = $(relatedTarget).parents(`.${DX_MENU_CLASS}`)[0];
+        const $contextMenuTarget = $(relatedTarget).parents(`.${DX_CONTEXT_MENU_CLASS}`)[0];
```

Comment:
I think the code will be more intuitive if we write these checks in the following manner:
```javascript
    const isInsideRootMenu =  $(relatedTarget).parents(`.${DX_MENU_CLASS}`).length !== 0;
   ...
```

Reply:
[3159bd5f6586777260033510f1278c6b4ecf6aea]